invoices/invoiceregistry_test.go

```diff
@@ -678,3 +691,125 @@ type mockPayload struct {
 func (p *mockPayload) MultiPath() *record.MPP {
 	return p.mpp
 }
+
+func (t *testContext) tickAfter(duration time.Duration) <-chan time.Time {
+	t.timeLock.Lock()
+	defer t.timeLock.Unlock()
+
+	triggerTime := t.now.Add(duration)
+	log.Debugf("tickAfter called: duration=%v, trigger_time=%v",
+		duration, triggerTime)
+
+	c := make(chan time.Time)
+
+	chans := t.chans[triggerTime]
+	chans = append(chans, c)
+	t.chans[triggerTime] = chans
+
+	return c
+}
+
+func (t *testContext) setTime(now time.Time) {
+	t.timeLock.Lock()
+	defer t.timeLock.Unlock()
+
+	t.now = now
+	remainingChans := make(map[time.Time][]chan time.Time)
+	for t, chans := range t.chans {
+		if t.After(now) {
+			remainingChans[t] = chans
+			continue
+		}
+
+		for _, c := range chans {
+			c <- now
+		}
+	}
+
+	t.chans = remainingChans
+}
+
+// TestSettleMpp tests settling of an invoice with multiple partial payments.
+func TestSettleMpp(t *testing.T) {
+	ctx := newTestContext(t)
+	defer ctx.cleanup()
+
+	ctx.now = testTime
+
+	// Add the invoice.
+	_, err := ctx.registry.AddInvoice(testInvoice, hash)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	mppPayload := &mockPayload{
+		mpp: record.NewMPP(testInvoiceAmt, [32]byte{}),
+	}
+
+	// Send htlc 1.
+	hodlChan1 := make(chan interface{}, 1)
+	event, err := ctx.registry.NotifyExitHopHtlc(
+		hash, testInvoice.Terms.Value/2,
+		testHtlcExpiry,
+		testCurrentHeight, getCircuitKey(10), hodlChan1, mppPayload,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if event != nil {
+		t.Fatal("expected no direct resolution")
+	}
+
+	// Wait for timer to be set up.
+	time.Sleep(100 * time.Millisecond)
```

Comment:
could have a channel signal it is ready instead

Reply:
modified so that tick channels also tick immediately when the trigger time is already there.